cpp/include/rapidsmpf/buffer/buffer.hpp

```diff
@@ -208,6 +226,8 @@ class Buffer {
     /// @brief The underlying storage host memory or device memory buffer (where
     /// applicable).
     StorageT storage_;
+    /// @brief CUDA event used to track copy operations
+    std::unique_ptr<Event> event_;
```

Comment:
Didnt we all agree that this needs to be a shared_ptr? Otherwise, for all user provided buffers, we will have to create an individual `Event`. Which is unnecessary IMO.

Reply:
Yes, this was my mistake. Fixed in https://github.com/rapidsai/rapidsmpf/pull/225/commits/dc1d89854e0f786c15fd6c0c1fe7d38ce8734943.